src/utils/hooks/useForceProjectSelection.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom-v5-compat';
+
+import { DEFAULT_NAMESPACE } from '@kubevirt-utils/constants/constants';
+import { useIsAdmin } from '@kubevirt-utils/hooks/useIsAdmin';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { ALL_NAMESPACES_PATH } from '@virtualmachines/tree/hooks/constants';
+import { VIRTUALIZATION_PATHS } from '@virtualmachines/tree/utils/constants';
+
+import useProjects from './useProjects';
+
+type UseForceProjectSelectionOptions = {
+  enabled?: boolean;
+  pathConditions?: string[];
+};
+
+export const useForceProjectSelection = (options: UseForceProjectSelectionOptions = {}) => {
```

Comment:
with only two parameters, instead of using the object i would rather use two distinct parameters pathConditions and enable that can also have default values `[VIRTUALIZATION_PATHS.BASE]` and `true`